xarray/core/dataarray.py

```diff
@@ -110,6 +111,31 @@ def __setitem__(self, key, value):
         self.data_array[pos_indexers] = value
 
 
+class DataArraySizes(collections.Mapping, formatting.ReprMixin):
```

Comment:
Probably not -- could just generate an object on the fly, e.g., `Frozen(OrderedDict(zip(self.dims, self.shape)))`. Let me do a quick micro benchmark to see if that has appreciably different performance.